### DISCUSSION TITLE
How to configure TrackingServer's logs (NOT Log experiment data)

### DISCUSSION THREAD

**Discussion by NWTRP** [mlflow/mlflow#8108] [Q&A]
I'm attempting to diagnose issues with an AWS deployed container instance of MLFlow's tracking server.

To ensure I've not misconfigured a database, s3 bucket or other system I'd like to see output from a logger such as this one: https://github.com/mlflow/mlflow/blob/062aa18bd3c6c734447efd79eb55d030ec68d9c4/mlflow/cli.py#L35

Is there anyway to configure or pass an argument to the command line that will allow me to see or capture these logs?

Unfortunately all the documentation regarding logging in MLFlow refers to logging experimental data rather than that of the process itself further complicating the matter.

In my docker I am running:

CMD ["pipenv", "run", "mlflow", "server", "-h", "0.0.0.0", "-p", "43000", \
  "--serve-artifacts", "--default-artifact-root", "s3://my-bucket/mlflow", \
  "--artifacts-destination", "s3://my-bucket/mlflow/artifacts", \
  "--gunicorn-opts", "--log-level=debug --ca-certs=my-ca-certificates.crt"]
Which will log out information from gunicorn (i.e. web-server requests), however no MLFlow specific logging, error capture and trace output is shown.

Given that MlFlow is littered with logging statements, I would hope to see output from some of these statements when the server initialises: https://github.com/mlflow/mlflow/search?q=_logger

**Comment by JHibbard** [comment]
Hi NWTRP,

That sounds frustrating. Is the stderr and stdout of your MLflow server both being redirected to your container's stderr and stdout?

Additional links that might help get to a solution:
- [logging_utils](https://github.com/mlflow/mlflow/blob/5224ad10cd65853354d81a75ba61c9b0ca94c9bf/mlflow/utils/logging_utils.py)
- [specify app name for logger](https://github.com/mlflow/mlflow/pull/7982)
